edx_django_utils/db/queryset_utils.py

```diff
@@ -0,0 +1,49 @@
+"""
+Utils related to QuerySets.
+"""
+
+
+def chunked_queryset(queryset, chunk_size=2000):
+    """
+    Slice a queryset into chunks.
+
+    The function slices a queryset into smaller QuerySets containing chunk_size objects and then yield them.It is used
+    to avoid memory error when processing huge queryset, and also database error due to that the database pulls whole
+    table at once. Concurrent database modification wouldn't make some entries repeated or skipped in this process.
```

Comment:
Here are some minor fixes.
```suggestion
    The function slices a queryset into smaller QuerySets containing chunk_size objects and then yields them. It is
    used to avoid memory error when processing huge querysets, and also to avoid database errors due to the
    database pulling the whole table at once. Concurrent database modification wouldn't make some entries
    repeated or skipped in this process.
```

However, I'm not clear on the meaning of the last sentence, so I'm not sure how to fix it.  Are you saying that without this function, another problem is that concurrent database modifications will cause some entries to be repeated or skipped? If so, maybe use:
```
Additionally, without using a chunked queryset, concurrent database modification while processing a large table might repeat or skip some entries.
```

Or, is this a problem that is introduced by chunked queryset?

Reply:
Yes, Without using this function, concurrent database modifications will cause some entries to be repeated or skipped, due to fetching all the data from the database at once.
I have added one more test case to support this case.